.github/workflows/codecov.yml

```
@@ -16,10 +16,15 @@ jobs:
         with:
           go-version: '1.22'
           check-latest: true
-      - name: Optional init
-        run: make dependencies || true # Used in fortio for instance to prep for go tests
       - name: Run test coverage
-        run: go test -coverprofile=coverage.out ./...
+        run: |
+          if make -n coverage &>/dev/null; then
+            echo "Running coverage through make"
+            make coverage
+          else
+            echo "No Makefile coverage target, using default behavior"
+            go test -coverprofile=coverage.out ./...
```

I would recommend using -race there


```suggestion
            go test -race -coverprofile=coverage.out ./...
```

It helps to detect things

---

regular non coverage tests are already running with -race and there is a reason coverage doesn't :) [to get more coverage]

---

https://github.com/fortio/workflows/blob/main/.github/workflows/gochecks.yml#L39

---

Indeed.

But I'm a bit surprised you are launching test twice then, once with coverage and once without

---

one is functional test and the other a coverage test (obviously both will fail if there is some basic test failure but the actions run in parallel and are free to run) and again it does need to be run at least twice (and in log's new case... 4 times with combos of build tags)